### PR TITLE
Cleaned up map API. Note breaking changes.

### PR DESCRIPTION
**Pull request by oscar-nutonomy**
This cleans up the MAP API.

1) Remove accumulative dilation behavior.
2) Uses caching library
3) Renames `get_pixel` to `to_map_coord`. This now calls `transform_matrix` instead of re-implementing logic.
4) Supports different resolution on init. This helps clean up the Explorer code.

TODO:
* [x] Add tests.
* [ ] Update to load new binary format, use 1 for foreground and 0 for background.

### CODE REVIEW

**Comment by oscar-nutonomy** [comment]
@sourabh-nutonomy @Alex-nutonomy : updated API per discussion.

File: python-sdk/nuscenes/utils/map_mask.py [hunk 1]
```diff
@@ -14,109 +16,104 @@
...
-        on_mask[py >= self.mask.shape[0]] = False
+        on_mask[py >= this_mask.shape[0]] = False
 
-        on_mask[on_mask] = (self.mask[py[on_mask], px[on_mask]] == self.foreground)
+        on_mask[on_mask] = this_mask[py[on_mask], px[on_mask]] == self.foreground
 
         return on_mask
 
-    def get_pixel(self, x: np.array, y: np.array) -> Tuple[np.ndarray, np.ndarray]:
+    def to_map_coord(self, x, y) -> Tuple[np.ndarray, np.ndarray]:
```
Comment (sourabh-nutonomy) [inline python-sdk/nuscenes/utils/map_mask.py]: The name should be updated. It actually returns the pixel coordinates in the map image and not the global map coordinates. The docstring should be updated as well.

File: python-sdk/tutorial.ipynb [hunk 2]
```diff
@@ -403,9 +403,9 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "nuenv",
```
Comment (sourabh-nutonomy) [inline python-sdk/tutorial.ipynb]: Revert these changes probably. In the virtual environment installation instructions, we name the environment as `nuscenes`.

**Comment by oscar-nutonomy** [comment]
@sourabh-nutonomy : addressed. Also added tests. Please check again.

**Comment by sourabh-nutonomy** [comment]
@oscar-nutonomy Looks good to me. As discussed, I will convert the maps to binary after we unify the maps for each region.
